SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/BaseData.cpp

```diff
@@ -139,28 +135,22 @@ bool BaseData::setParent(BaseData* parent, const std::string& path)
 
         m_counter++;
         m_isSet = true;
-    }
+    }else if (!path.empty())
```

Comment:
what if parent is null and path empty. Aka wrong use of this method ?
maybe a
`` else
{
msg_ warning 
return false;
}```
?